### PR TITLE
[PROF-10486] Update docs to mark Ruby allocation profiling GA

### PR DESCRIPTION
**Pull request by ivoanjo**
### What does this PR do? What is the motivation?

This PR updates the documentation to mark the Ruby allocation profiling feature as GA as of library version 2.3.0.

### Merge instructions

- [x] Please merge after reviewing

### Additional notes

I decided to do a pass on all Ruby tidbits in the documentation and ended up collecting a few more cleanups / improvements / simplifications.

### CODE REVIEW

File: content/en/profiler/connect_traces_and_profiles.md [hunk 1]
```diff
@@ -287,8 +284,6 @@ Requires `dd-trace-go` version 1.37.0+ and works best with Go version 1.18 or ne
 
 Endpoint profiling is enabled by default when you [turn on profiling for your Ruby service][1].
 
-Requires `dd-trace-rb` version 0.54.0+.
-
```
Comment (ivoanjo) [inline content/en/profiler/connect_traces_and_profiles.md:291]: This version is extremely old, so this is no longer relevant.

File: content/en/profiler/enabling/supported_versions.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ To use the Datadog Profiler, use at least the minimum versions summarized in the
 
 |                                   |  [Java][1]   |   [Python][2]    |    [Go][3]    |   [Ruby][4]    | [Node.js][5]  |  [.NET][6]  |   [PHP][7]    | [Rust/C/C++][8] |
 |-----------------------------------|:------------:|:----------------:|:-------------:|:--------------:|:-------------:|:-----------------------------------------------------------------------:|:-------------:|:---------------:|
-| <strong>Minimum&nbsp;runtime&nbsp;version</strong> | [JDK&nbsp;8+][17]  | Python&nbsp;2.7+ | Go&nbsp;1.19+ | Ruby&nbsp;2.3+ | Node&nbsp;14+ | .NET&nbsp;Core&nbsp;2.1+, .NET&nbsp;5+, .NET&nbsp;Framework&nbsp;4.6.1+ | PHP&nbsp;7.1+ |                 |
+| <strong>Minimum&nbsp;runtime&nbsp;version</strong> | [JDK&nbsp;8+][17]  | Python&nbsp;2.7+ | Go&nbsp;1.19+ | Ruby&nbsp;2.5+ | Node&nbsp;14+ | .NET&nbsp;Core&nbsp;2.1+, .NET&nbsp;5+, .NET&nbsp;Framework&nbsp;4.6.1+ | PHP&nbsp;7.1+ |                 |
```
Comment (ivoanjo) [inline content/en/profiler/enabling/supported_versions.md:23]: This was outdated -- with library version 2.0 (released in early June) we require Ruby 2.5+

File: content/en/profiler/profiler_troubleshooting/ruby.md [hunk 3]
```diff
@@ -13,8 +13,8 @@ further_reading:
 
 If you've configured the profiler and don't see profiles in the profile search page, turn on [debug mode][1] and [open a support ticket][2] with debug files and the following information:
 
-- Operating system type and version (for example, Ubuntu Linux 22.04)
-- Runtime type, version, and vendor (for example, Ruby 2.7.3)
+- Operating system type and version (for example, Ubuntu Linux 24.04)
+- Runtime type, version, and vendor (for example, Ruby 3.3.1)
```
Comment (ivoanjo) [inline content/en/profiler/profiler_troubleshooting/ruby.md:17]: Refreshed :)

File: content/en/profiler/profiler_troubleshooting/ruby.md [hunk 4]
```diff
@@ -30,7 +30,7 @@ There is a known incompatibility between Ruby 2.7 and older GCC versions (4.8 an
 
 To fix this, update your operating system or Docker image so that the GCC version is something more recent than v4.8.
 
-For further help with this issue, [contact support][2] and include the output of running `DD_PROFILING_FAIL_INSTALL_IF_MISSING_EXTENSION=true gem install ddtrace` and the resulting `mkmf.log` file.
+For further help with this issue, [contact support][2] and include the output of running `DD_PROFILING_FAIL_INSTALL_IF_MISSING_EXTENSION=true gem install datadog` and the resulting `mkmf.log` file.
```
Comment (ivoanjo) [inline content/en/profiler/profiler_troubleshooting/ruby.md:33]: Updated for new library name :)

**Review by brett0000FF — APPROVED** [review]
Looks great--thank you @ivoanjo !

**Pull request merged**